### ISSUE TITLE
Enforce usage of object literal property value shorthand syntax

### ISSUE THREAD

**Issue by derekcicerone** [palantir/tslint#304]
See: http://ariya.ofilabs.com/2013/02/es6-and-object-literal-property-value-shorthand.html

**Comment by adidahiya** [comment]
+1 for a new rule that enforces this. It could be called `object-literal-shorthand`?

**Comment by ashwinr** [comment]
:+1: to both the rule and the name

**Comment by adidahiya** [comment]
We have another rule that operates on object literals called `sort-object-literal-keys`.
Perhaps that rule should be renamed to `object-literal-sorted-keys` to match this name?

**Comment by ashwinr** [comment]
sounds unnecessarily verbose.

**Comment by ashwinr** [comment]
actually, no. you're right, let's do that. I think `object-literal-sort-keys`

**Comment by adidahiya** [comment]
This rule should also handle functions in object literals.

good:

```ts
const iterableThing = {
    [Symbol.iterator]() {
        ...
    },
    next() {
        ...
    }
}
```

bad:

```ts
const iterableThing = {
    [Symbol.iterator]: function () {
        ...
    },
    next: function () {
        ...
    }
}
```

**Comment by danvk** [comment]
This is `object-shorthand` in eslint: http://eslint.org/docs/rules/object-shorthand
